pkg/kubectl/describe.go

```diff
@@ -439,6 +440,85 @@ func describePod(pod *api.Pod, rcs []api.ReplicationController, events *api.Even
 	})
 }
 
+func describeVolumes(volumes []api.Volume, out io.Writer) {
+	if volumes == nil || len(volumes) == 0 {
+		fmt.Fprint(out, "No volumes.\n")
+		return
+	}
+	fmt.Fprint(out, "Volumes:\n")
+	for _, volume := range volumes {
+		fmt.Fprintf(out, "  %v:\n", volume.Name)
+		if volume.VolumeSource.HostPath != nil {
```

Comment:
We need a multi-line if/else to determine which volume point is non-nil.  After that, a switch can be used to determine which printf statements to run.  That could be good, though, because each block of printf statements can be encapsulated into a function and re-used in https://github.com/kubernetes/kubernetes/pull/12733.   I copied the same set of printf statements in that PR.